### PR TITLE
runtime: remove randomness collective flip pallet

### PR DESCRIPTION
**Pull request by andresilva**
This is insecure and should not be used, in particular as we have BABE as a better source of randomness.

### CODE REVIEW

**Comment by andresilva** [comment]
This pallet stored the last 81 block hashes in its storage, should we bother with writing a runtime migration to clean that up?

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -1388,7 +1386,6 @@ construct_runtime! {
 	{
 		// Basic stuff; balances is uncallable initially.
 		System: frame_system::{Pallet, Call, Storage, Config, Event<T>} = 0,
-		RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 32,
```
Comment (ordian) [inline runtime/kusama/src/lib.rs:1391]: should it be replaced with a dummy that occupies the index or is it fine to overwrite it (in which case what's the point of the index?)?

**Comment by shawntabrizi** [comment]
> This pallet stored the last 81 block hashes in its storage, should we write a runtime migration to clean that up?

Only if it is being used in one of our production pallets

File: runtime/kusama/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:1391]: Nothing needs to occupy the index. The index is used to generate a unique identifier for the metadata.

**Review by shawntabrizi — APPROVED** [review]

File: runtime/kusama/src/lib.rs (hunk 1 above)
Comment (ordian) [reply runtime/kusama/src/lib.rs:1391]: To test my understanding:
1. Index is used to ensure stable identifier for the metadata when a new pallet is introduced (and not depend on the order of declaration).
2. It's fine to reuse an index later as long there is migration in place?

**Review by rphmeier — APPROVED** [review]
bless up

File: runtime/kusama/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:1391]: Yeah, assuming you clean up the old storage items, and that you update your tx version

**Comment by shawntabrizi** [comment]
@andresilva take a look at the migration. if all looks okay, you can merge

**Comment by andresilva** [comment]
bot merge

**Pull request merged**